### ISSUE TITLE
why    decay_steps   not  related  with  num_clones ？？

### ISSUE THREAD

**Issue by guicunbin** [tensorflow/models#3498]
https://github.com/tensorflow/models/blob/17a5bdc87855f5fc23ebc89baf5484f82c4698e2/research/slim/train_image_classifier.py#L236

here defined decay_steps,    it  may  be  a  error;
total_batch_size = FLAGS.num_clones * FLAGS.batch_size; 
So  here  may  be:     
decay_steps = int(num_samples_per_epoch / (FLAGS.batch_size  *  FLAGS.num_clones  )* 
              FLAGS.num_epochs_per_decay);

@nealwu   @sguada   @tfboyd      Is that right     ？？？

**Comment by angersson** [comment]
I can't understand what you're asking. Can you rephrase the question? Thanks!

**Comment by guicunbin** [comment]
@tfboyd  say that   
one  GPU  process  FLAGS.batch_size  images   
FLAGS.num_clones  GPU  will  process     FLAGS.batch_size * FLAGS.num_clones   images ,
so the number  in one epoch  will  be : 
num_samples_per_epoch / (FLAGS.batch_size * FLAGS.num_clones )
So  
decay_steps = decay_steps = int(num_samples_per_epoch / (FLAGS.batch_size * FLAGS.num_clones )*
FLAGS.num_epochs_per_decay);
is  that right ? @angersson

**Comment by tfboyd** [comment]
I am not sure what you are suggesting I said.  A snippet of the command I would run to run a batch-size of 64 per GPU for a total batch of 64*8 would be
```
   --num_clones=8 \
   --optimizer=sgd \
   --batch_size=64 \
```
I think what you are saying is that translation of epochs to steps ends up incorrect because the equation does not account for number of GPUs.  To make it more concrete I assume ImageNet is 1.2M images.
```
(1,200,000 / 64) * 2 (how many epochs to wait before the decay starts) = 37,500 global steps
# The problem you are suggesting is that in reality you have do 64*8 images per global step and after 37,500 steps you have done 16 epochs not 2 and instead should have done
((samples_per_epoch / (64*8) )*2 - 4,687 steps.
```
The code says it is global_steps so the mistake you are saying seems accurate, but I do not have any direct experience to know if there is something I am missing.  I think I have more clearly conveyed your concern, but I can not answer your question with confidence.  

If you are training ResNet (not saying you are) you should check out the [updated official model ](https://github.com/tensorflow/models/tree/master/official/resnet) that has less confusion.  We are trying to move away from these SLIM examples but have not converted enough samples to do so.  We have a very easy Multi-GPU coming out in the next few weeks and an early version of it as a PR for the code in the branch that requires using tensorflow-nightly.  A bit bleeding edge but we know the model converges and the approach is much easier to implement (2-3 lines of code and soon 1).

**Comment by karmel** [comment]
Closing this as the Official Model now has multi-GPU support, and further updates are coming to that over the next few weeks.
